### ISSUE TITLE
ResteasyReactiveProcessor#setupEndpoints reports duplicate endpoints when a rest client matches a resource

### ISSUE THREAD

**Issue by DeMol-EE** [quarkusio/quarkus#44190]
### Describe the bug

My project defines a resource (GET /hello) and a rest client to a remote service, which happens to have a similar resource (GET /hello). When running tests, ResteasyReactiveProcessor throws jakarta.enterprise.inject.spi.DeploymentException, as it believes that rest client is a duplicate of my own resource.

### Expected behavior

No exception.

### Actual behavior

jakarta.enterprise.inject.spi.DeploymentException is thrown.

### How to Reproduce?

Clone the [reproducer](https://github.com/DeMol-EE/quarkus-reproducer-44190) and run `mvn test`.

### Output of `uname -a` or `ver`

Darwin 03ccaf0a-638d-4f7e-9e35-498cd06fb36a.fritz.box 24.1.0 Darwin Kernel Version 24.1.0: Thu Oct 10 21:03:15 PDT 2024; root:xnu-11215.41.3~2/RELEASE_ARM64_T6000 arm64

### Output of `java -version`

OpenJDK 64-Bit Server VM Temurin-21.0.4+7 (build 21.0.4+7-LTS, mixed mode)

### Quarkus version or git rev

3.15.1

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.9.8 (36645f6c9b5079805ea5009217e36f2cffd34256)

**Comment by DeMol-EE** [comment]
It’s worth pointing out that running the application (dev or prod mode) works fine.

**Comment by geoand** [comment]
Thanks for reporting!

I'll have a look soon

**Comment by DeMol-EE** [comment]
Thanks! On a related note: I suspect this might also be causing rest client endpoints to show up in the generated OpenAPI file (when also using quarkus-smallrye-openapi). I think I saw a few posts about this, but none of them ending in anything because it turned out to be hard to reproduce. Probably because the endpoints only show up when the file is generated when the test profile is active. This happens to be exactly something that we do in one of our projects as well (we have some test that detects OpenAPI changes), and we also observe that the rest client endpoints appear in our generated file (though we have also added a workaround using OASFilter to filter them out again).
